src/main/java/uk/gov/hmcts/befta/dse/ccd/definition/converter/EnvironmentURLUtils.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.befta.dse.ccd.definition.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import uk.gov.hmcts.befta.exception.InvalidPropertyException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class EnvironmentURLUtils {
+
+    private static final String BASE_URL_PLACEHOLDER_REGEX = "\\$\\{(.*?)}";
+    private static final List<String> SHEETS_FOR_URL_SUBSTITUTIONS = Arrays.asList("CaseEvent", "CaseEventToFields");
+
+    public static JsonNode updateCallBackURLs(JsonNode rootSheetArray, String jsonFileName)
+            throws JsonProcessingException, MalformedURLException, InvalidPropertyException {
+        if (SHEETS_FOR_URL_SUBSTITUTIONS.contains(jsonFileName)) {
+            return new ObjectMapper().readTree(parseCallbackHostValues(rootSheetArray.toString()));
+        } else {
+            return rootSheetArray;
+        }
+    }
+
+    private static String parseCallbackHostValues(String sheet) throws MalformedURLException, InvalidPropertyException {
+        Matcher matcher = Pattern.compile(BASE_URL_PLACEHOLDER_REGEX).matcher(sheet);
+
+        while (matcher.find()) {
+            BaseUrlPlaceholder matchedResult = parseDefaultHostValue(matcher);
+
+            String replacementUrl = getURLStringFromEnvironmentValue(matchedResult.getEnvironmentVariable());
+            if (replacementUrl == null) {
+                replacementUrl = matchedResult.getUrlDefaultValue();
+            }
+
+            sheet = sheet.replace(matcher.group(0), replacementUrl);
+        }
+
+        return sheet;
+    }
+
+    private static String getURLStringFromEnvironmentValue(String environmentName) throws MalformedURLException {
+        String environmentValue = System.getenv(environmentName);
+        return environmentValue != null ? new URL(environmentValue).toString() : null;
+    }
+
+    private static BaseUrlPlaceholder parseDefaultHostValue(Matcher matcher) throws InvalidPropertyException {
+        String[] split = matcher.group(1).split(":", 2);
+
+        if (split.length != 2 || split[1].isEmpty()) {
+            throw new InvalidPropertyException(
```

Comment:
Could the name be InvalidPlaceholderException?

Reply:
https://github.com/hmcts/befta-fw/pull/123/commits/6d00f97fd256e7a129a7e535d4c70f0928eb64e9